doc/theory.md

```diff
@@ -52,3 +52,18 @@ Finally, FLOSS diffs the emulator memory states from before and
   6. Emulate decoder functions using extracted arguments and emulator state
   7. Diff memory state from before and after decoder emulation
   8. Extract human-readable strings from memory state difference
+
+## Go String Extraction
+The Go string extraction algorithm implemented in the FLOSS tool is designed to address the unique characteristics of Go strings in binaries, ensuring efficient retrieval during reverse engineering and malware analysis tasks. Go strings are conventionally represented as sequences of 8-bit bytes, often encoding UTF-8 text. Without special handling, the strings within the binary may appear as a large chunk of indistinguishable data. The FLOSS algorithm overcomes this challenge by analyzing struct string instances and leveraging the length-sorted order of strings within the binary. This capability provides valuable insights into the inner workings of Go programs, facilitating comprehensive analysis and understanding.
+
+### Algorithm:-
+
+1. Analyze the struct string instances within the binary.
```

Comment:
Should we describe struct strings in more detail? It's probably not clear to most people what these are.